oak-exercise/src/main/java/org/apache/jackrabbit/oak/exercise/security/authorization/models/simplifiedroles/Role.java

```diff
@@ -44,7 +49,9 @@ private Role(long permissions, String... privilegeNames) {
 
     private Role(@NotNull Role base, long permissions, String... privilegeNames) {
         this.permissions = base.permissions|permissions;
-        this.privilegeNames = ImmutableSet.<String>builder().addAll(base.privilegeNames).add(privilegeNames).build();
+        this.privilegeNames = Collections.unmodifiableSet(
```

Comment:
private constructor called from within the class, `privilegeNames` can't be null here.